packages/wonder-blocks-core/src/components/render-state-root.js

```diff
@@ -0,0 +1,34 @@
+// @flow
+import * as React from "react";
+
+// TODO(somewhatabstract, FEI-4174): Update eslint-plugin-import when they
+// have fixed:
+// https://github.com/import-js/eslint-plugin-import/issues/2073
+// eslint-disable-next-line import/named
+import {RenderState, RenderStateContext} from "./render-state-context.js";
+
+const {useEffect, useRef, useState} = React;
+
+type Props = {|
+    children: React.Node,
+|};
+
+export const RenderStateRoot = ({children}: Props): React.Node => {
+    const initialRef = useRef<boolean>(true);
+    const [, setMounted] = useState<boolean>(false);
```

Comment:
Maybe it'd be worth moving this to a `useForceUpdate` hook or something like that? Mostly for making it easy to understand what this does, rather than actually sharing the code...

Reply:
If I had to encapsulate this within a hook, I'd probably call it something like `useRerenderOnce` or something like that.  I'd expect `useForceUpdate(bool)` to trigger an update whenever `bool` is `true`.  I can put it in a hook defined locally in this file and document it better, but I have my doubts as to how useful this chunk of code will be outside of `wonder-blocks-core`.

Reply:
Ah that's a good point. I don't expect it to be useful, I was just thinking it might make it more obvious what the intention is here. But I guess we could also just add a comment, or maybe even name `setMounted` to something else, like `forceRerender`?

Reply:
I'll add comments.  Having `setMounted` without using `mounted` is a bit weird and better naming could help.

Reply:
I think I can simplify this by replacing the `useRef` and `useState` with a single `useState.